easybuild/easyconfigs/c/CP2K/CP2K-5.1-intel-2017b.eb

```diff
@@ -0,0 +1,42 @@
+name = 'CP2K'
+version = '5.1'
+
+homepage = 'http://www.cp2k.org/'
+description = """CP2K is a freely available (GPL) program, written in Fortran 95, to perform atomistic and molecular
+ simulations of solid state, liquid, molecular and biological systems. It provides a general framework for different
+ methods such as e.g. density functional theory (DFT) using a mixed Gaussian and plane waves approach (GPW), and
+ classical pair and many-body potentials. """
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+toolchainopts = {'pic': True, 'precise': True}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TAR_BZ2]
+patches = [
+    'CP2K-2.4.0-fix_compile_date_lastsvn.patch',
+]
+checksums = [
+    'e23613b593354fa82e0b8410e17d94c607a0b8c6d9b5d843528403ab09904412',  # cp2k-5.1.tar.bz2
+    '02475cbe24c8d4ba27037c826adf8a534cad634c3c4e02c21d743f5284516bda',  # CP2K-2.4.0-fix_compile_date_lastsvn.patch
+]
+
+dependencies = [
+    ('Libint', '1.1.6'),
+    ('libxc', '3.0.0'),
+    ('libxsmm', '1.8.1'),
+    ('FFTW', '3.3.6'),
+]
+
+builddependencies = [
+    ('flex', '2.6.4'),
+    ('Bison', '3.0.4'),
+]
+
+# values to use here are +1 those used for building Libint
+# see https://github.com/cp2k/cp2k/blob/master/cp2k/tools/hfx_tools/libint_tools/README_LIBINT
+extradflags = "-D__LIBINT_MAX_AM=6 -D__LIBDERIV_MAX_AM1=5"
```

Comment:
This is quite important, see also discussion in https://github.com/easybuilders/easybuild-easyblocks/issues/1174.

And maybe this needs to be leveraged to the CP2K easyblock instead, since you can query how LibInt was configured, I think...

Reply:
Yes, certainly possible, in line with the README_LIBINT:

```
$ ml load Libint/1.1.6-intel-2016b
$ grep LIBINT_MAX_AM $EBROOTLIBINT/include/libint/libint.h 
#define LIBINT_MAX_AM 5
$ grep LIBDERIV_MAX_AM1 $EBROOTLIBINT/include/libderiv/libderiv.h 
#define LIBDERIV_MAX_AM1 4
#define LIBDERIV_MAX_AM12 3
```

You need to increment these numbers by 1. (I guess CP2K does not pick up these definitions because the fortran interface to Libint does not define them.)